rpctest/utils.go

```diff
@@ -76,24 +77,25 @@ func syncMempools(nodes []*Harness) error {
 	return nil
 }
 
-// syncBlocks blocks until all nodes report the same block height.
+// syncBlocks blocks until all nodes report the same best chain.
 func syncBlocks(nodes []*Harness) error {
 	blocksMatch := false
 
+retry:
 	for !blocksMatch {
-	retry:
-		blockHeights := make(map[int64]struct{})
+		blocksByHash := make(map[chainhash.Hash]int32)
 
 		for _, node := range nodes {
-			blockHeight, err := node.Node.GetBlockCount()
+			blockHash, blockHeight, err := node.Node.GetBestBlock()
 			if err != nil {
 				return err
 			}
 
-			blockHeights[blockHeight] = struct{}{}
-			if len(blockHeights) > 1 {
+			existingHeight, ok := blocksByHash[*blockHash]
```

Comment:
Seems like the two-value read here is unnecessary. It's never used within the loop, and the `existingHeight` value can safely be used if the node's `blockHash` isn't found within the map since the value is simply an `int32` rather than a pointer.